### ISSUE TITLE
Sudden increase of loss (from 0.1 to thousands) after ~4000 steps

### ISSUE THREAD

**Issue by rchen1207** [lc222/char-cnn-text-classification-tensorflow#3]
I got a sudden drop of accuracy from ~90% to below 20% after about 4000 training steps? An answer on [stack overflow](https://stackoverflow.com/questions/42327543/adam-optimizer-goes-haywire-after-200k-batches-training-loss-grows/42420014#42420014) says that's a problem of Adam, and suggests a higher epsilon to fix?

Do you have any suggestion based on your training experience? Thanks!

My Stats:
```
...
2018-03-21T13:56:22.703288: step 3866, loss 0.43315, acc 0.90625
2018-03-21T13:56:34.078052: step 3867, loss 0.479325, acc 0.851562
2018-03-21T13:56:49.068364: step 3868, loss 0.256937, acc 0.9375
2018-03-21T13:57:06.444321: step 3869, loss 0.273636, acc 0.90625
2018-03-21T13:57:15.877391: step 3870, loss 0.232426, acc 0.929688
2018-03-21T13:57:23.503175: step 3871, loss 3.60959, acc 0.5625
2018-03-21T13:57:31.559549: step 3872, loss 8.54077, acc 0.46875
2018-03-21T13:57:41.647740: step 3873, loss 5.25972, acc 0.429688
2018-03-21T13:57:51.344488: step 3874, loss 322.297, acc 0.234375
2018-03-21T13:58:01.384422: step 3875, loss 21.8541, acc 0.453125
2018-03-21T13:58:10.741925: step 3876, loss 131.247, acc 0.242188
2018-03-21T13:58:19.820908: step 3877, loss 4977.22, acc 0.289062
2018-03-21T13:58:29.067122: step 3878, loss 155.665, acc 0.289062
2018-03-21T13:58:36.979536: step 3879, loss 1825.98, acc 0.226562
2018-03-21T13:58:44.365748: step 3880, loss 1705.01, acc 0.304688
2018-03-21T13:58:53.205122: step 3881, loss 1100.61, acc 0.25
2018-03-21T13:59:01.279594: step 3882, loss 12413.7, acc 0.234375
2018-03-21T13:59:10.017025: step 3883, loss 23248.9, acc 0.273438
...
```
![accuracy](https://user-images.githubusercontent.com/11575904/37698105-e85587c6-2d1b-11e8-8556-f20beef57232.png)
